thirdparty/boost/CMakeLists.txt

```diff
@@ -0,0 +1,121 @@
+message(STATUS "Building Boost ${_acts_boost_version}")
+
+include(ExternalProject)
+
+string(REPLACE "." "_" _acts_boost_version_ ${_acts_boost_version})
+set(Boost_URL "https://boostorg.jfrog.io/artifactory/main/release/${_acts_boost_version}/source/boost_${_acts_boost_version_}.tar.gz")
+set(Boost_LIBRARY_LIST "filesystem,program_options,test")
+set(Boost_SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/boost-prefix/src/boost)
+set(Boost_INCLUDE_DIR ${Boost_SOURCE_DIR})
+set(Boost_LIBRARY_DIR ${Boost_SOURCE_DIR}/stage/lib)
+set(Boost_B2_CXXFLAGS -std=c++17\ -fPIC\ -O3)
+set(Boost_B2_FLAGS variant=release threading=multi runtime-link=shared link=shared cxxflags=${Boost_B2_CXXFLAGS})
+
+# Target for boost headers
+
+# Note: Since all builds below depend on the headers, use this step to
+# download the boost release
+
+ExternalProject_Add(
+  boost
+  URL ${Boost_URL}
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND ""
+  INSTALL_COMMAND ""
+)
+
+add_library(Boost::boost INTERFACE IMPORTED GLOBAL)
+add_dependencies(Boost::boost boost)
+target_include_directories(Boost::boost INTERFACE ${Boost_INCLUDE_DIR})
+install(DIRECTORY ${Boost_INCLUDE_DIR}/boost DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+# Target for bootstrap step
+
+# Note: Required by all builds but not by header-only target, so need
+# a separate target
+
+# Note: Re-use ExternalProject_Add instead of Add_Step because I can't
+# figure out how to avoid race conditions and duplicate downloads with
+# Add_Step targets.  From cmake's point of view they will all be
+# different project but they all use the same working directory, so it
+# does the right thing.  (L-G)
+
+ExternalProject_Add(
+  boost_bootstrap
+  DOWNLOAD_COMMAND ""
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND ./bootstrap.sh --with-libraries=${Boost_LIBRARY_LIST}
+  INSTALL_COMMAND ""
+  DEPENDS boost
+  LOG_BUILD true
+  SOURCE_DIR ${Boost_SOURCE_DIR}
+  BUILD_IN_SOURCE true
+  EXCLUDE_FROM_ALL true
+)
+
+# Target for Boost::filesystem
+
+ExternalProject_Add(
+  boost_build_filesystem
+  DOWNLOAD_COMMAND ""
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND ./b2 ${Boost_B2_FLAGS} --with-filesystem
+  INSTALL_COMMAND ""
+  DEPENDS boost_bootstrap
+  SOURCE_DIR ${Boost_SOURCE_DIR}
+  LOG_BUILD true
+  BUILD_IN_SOURCE true
+  BUILD_BYPRODUCTS ${Boost_LIBRARY_DIR}/libboost_filesystem.so.${_acts_boost_version}
+  EXCLUDE_FROM_ALL true
+)
+
+add_library(boost_filesystem SHARED IMPORTED GLOBAL)
+set_target_properties(boost_filesystem PROPERTIES IMPORTED_LOCATION ${Boost_LIBRARY_DIR}/libboost_filesystem.so.${_acts_boost_version})
+add_dependencies(boost_filesystem boost_build_filesystem)
+add_library(Boost::filesystem ALIAS boost_filesystem)
```

Comment:
This target is note recovered in `ActsConfig.cmake` is it? That's what I meant in the other comment. Correct me if I'm wrong please!

Reply:
Correct, the shared libs are not exported, even tho they are installed. Let me know if you want them exported as well, it's an easy fix.